src/main/java/org/geysermc/discordbot/commands/moderation/ForumPostCommand.java

```diff
@@ -0,0 +1,355 @@
+            for (ForumTag forumTag : currentTags) {
+                if (forumTag.getName().equalsIgnoreCase(tag)) {
+                    tagFound = forumTag;
+                    break;
+                }
+            }
+
+            if (tagFound != null) {
+                List<ForumTagSnowflake> updatedTags = new ArrayList<>(currentTags);
+                updatedTags.remove(ForumTagSnowflake.fromId(tagFound.getId()));
+                threadChannel.getManager()
+                    .setAppliedTags(updatedTags)
+                    .queue(
+                        unused -> event.reply("Tag " + tag + " removed.").queue(),
+                        error -> event.reply("Could not remove tag from post").queue()
+                    );
+
+            } else {
+                event.reply("Post was not tagged with " + tag + ".").queue();
+            }
+        }
+
+        @Override
+        public void onAutoComplete(CommandAutoCompleteInteractionEvent event) {
+            // Get the query
+            String query = event.getFocusedOption().getValue();
+
+            // Get the tags
+            List<ForumTag> tags = potentialTags(event.getChannel().asThreadChannel().getAppliedTags(), query);
+
+            event.replyChoices(tags.stream()
+                    .distinct()
+                    .map(tag -> new Command.Choice(tag.getName(), tag.getName()))
+                    .limit(25)
+                    .toArray(Command.Choice[]::new))
+                .queue();
+        }
+    }
+
+    public static class CloseOldPostSubCommand extends SlashCommand {
+        public CloseOldPostSubCommand() {
+            this.name = "close-old";
+            this.help = "Close old posts";
+            this.userPermissions = new Permission[] { Permission.MESSAGE_MANAGE };
+            this.options = Collections.singletonList(new OptionData(OptionType.INTEGER, "days", "The minimum age in days of posts that will be closed in bulk", true));
+        }
+
+        @Override
+        protected void execute(@NotNull SlashCommandEvent event) {
+            int days = Objects.requireNonNull(event.getOption("days")).getAsInt();
+            Checks.notNull(event.getGuild(), "server");
+
+            ForumChannel forumChannel = ServerSettings.getForumChannel(event.getGuild());
+            if (forumChannel == null) {
+                event.reply("Forum channel not found.").queue();
+                return;
+            }
+
+            for (ThreadChannel channel : forumChannel.getThreadChannels()) {
+                if (channel.isArchived() || channel.isLocked() || channel.isPinned()) {
+                    continue;
+                }
+                OffsetDateTime channelCreated = channel.getTimeCreated();
+                OffsetDateTime closeTime = OffsetDateTime.now().minusDays(days);
+                if (channelCreated.isBefore(closeTime)) {
+                    ThreadChannelManager manager = channel.getManager().setArchived(true);
```

Comment:
There's no need to instantiate a variable here, we can just run `channel.getManager().setArchived(true).queue();`